libsolidity/codegen/CompilerContext.cpp

```diff
@@ -560,7 +560,11 @@ void CompilerContext::optimizeYul(yul::Object& _object, yul::EVMDialect const& _
 
 string CompilerContext::revertReasonIfDebug(string const& _message)
 {
-	return YulUtilFunctions::revertReasonIfDebug(m_revertStrings, _message);
+	return YulUtilFunctions::revertReasonIfDebugBody(
+		m_revertStrings,
+		"mload(" + to_string(CompilerUtils::freeMemoryPointer) + ")",
```

Comment:
Why can't we use allocate unbounded here?

Reply:
Because this will be used for ``CompilerContext::appendInlineAssembly`` and we cannot call Yul functions from there, at least not easily.